File: Quivi.Backoffice.Api/Requests/Availabilities/PatchAvailabilityRequest.cs
using Quivi.Backoffice.Api.Dtos;

namespace Quivi.Backoffice.Api.Requests.Availabilities
{
    public class PatchAvailabilityRequest : ARequest
    {
        public string? Name { get; init; }
        public bool? AutoAddNewMenuItems { get; set; }
        public bool? AutoAddNewChannelProfiles { get; set; }
        public IEnumerable<WeeklyAvailability>? WeeklyAvailabilities { get; init; }
    }
}

File: quivi.backoffice.react.webapp/src/hooks/api/Dtos/availabilities/PatchAvailabilityRequest.ts
import { WeeklyAvailability } from "./WeeklyAvailability";

export interface PatchAvailabilityRequest {
    readonly id: string;
    readonly name?: string;
    readonly autoAddNewChannelProfiles?: boolean;
    readonly autoAddNewMenuItems?: boolean;
    readonly weeklyAvailabilities?: WeeklyAvailability[];
}